### ISSUE TITLE
Can't resize below 10%

### ISSUE THREAD

**Issue by fidalgodev** [bvaughn/react-resizable-panels#162]
Hello 👋🏼 ,

First, thanks for a great library! I think this was requested before, but would be nice to instead of collapse the panel to be "invisible", it would be nice to collapse to a specific width. So maybe a `collapsibleSize` would be cool.

What I want to do, is when I'm resizing the panel, and it reaches a certain width, I want it to colapse do a specific width. This is so I have a sidebar with text that you can resize, and after a certain size it will collapse into a sidebar with just icons.

I'm using the workaround you mentioned [here](https://github.com/bvaughn/react-resizable-panels/issues/97) when a user asked for the same, but seems like I can't resize bellow 10%?

```js
const sidebarRef = useRef(null)
  const SIDEBAR_MIN_SIZE = 15

  const collapseSidePanel = () => {
    console.log('COLLAPSE')
    const sidebarPanel = sidebarRef.current
    if (sidebarPanel) {
      sidebarPanel.resize(2)
    }
  }

  const onSidebarResize = size => {
    if (size < SIDEBAR_MIN_SIZE) {
      collapseSidePanel()
    }
  }

 <PanelGroup direction="horizontal" autoSaveId="panel-sizes">
  <Panel ref={sidebarRef} onResize={onSidebarResize} defaultSize={20} maxSize={40}>
    <Navbar />
  </Panel>
  <StyledPanelResizeHandle/>
  <Panel defaultSize={80}>
    <Box as="main" flex="1">
      {children}
    </Box>
  </Panel>
</PanelGroup>
```

When I go narrow than `SIDEBAR_MIN_SIZE`, the `collapseSidePanel` is being called and I can see the `COLLAPSE` log, but at the same time I can see the `console.log('RESIZED', size)` log, and it says `10`, even tho I'm asking it to resize to `2` since I do `sidebarPanel.resize(2)`.

If I look at the DOM, even tho I call `sidebarPanel.resize(2)`, I can see that the `flex` gets set to: `flex: 10 1 0px;`. Anything above 10 is working fine, but anything bellow 10 doesn't seem to work.

**Comment by bvaughn** [comment]
I don't think the `collapsibleSize` feature you mentioned is something I want to build/support, but if you'll provide a repro that I can run (like a Code Sandbox) I can take a look and see if I can spot why the 10% thing is happening.

**Comment by fidalgodev** [comment]
> 

Here you go:
https://codesandbox.io/p/sandbox/relaxed-water-32swtr

There is 2 issues:
- Even tho the `resize()` is called with 2, it's being set to `10`
- Even tho we call resize and it's being set to `10`, the panel never changes size actually. You can see that when you go narrow than `SIDEBAR_MIN_SIZE`, we call `collapseSidePanel` which calls `sidebarPanel.resize(2);` and it seems to resize it, but because we are dragging it(i think), it never changes size.

I notice that when we use `collapsible`, all it does is set `flex: 0 0` on the panel. So it would be really cool to have a new prop called `collapsibleSize` or something, so we could just do:

`<Panel collapsible collapsibleSize={3} defaultSize={35} minSize={30} maxSize={40}>`

So it would collapse bellow `30` in this case, but instead of collapse to `0`, it would collapse to `3`.

I tried to achieve the same with the sandbox code but doesn't seem to be possible.

**Comment by fidalgodev** [comment]
Also tried this, but doesn't seem to work either:

```js
const sidebarRef = useRef(null)

const onCollapse = collapsed => {
  const sidebarPanel = sidebarRef.current
  if (sidebarPanel && collapsed) {
    console.log('got here, collapsed and resize')
    sidebarPanel.resize(10)
  }
}

<StyledPanel
  ref={sidebarRef}
  collapsible
  onCollapse={onCollapse}
  defaultSize={40}
  minSize={30}
  maxSize={50}
  >
  <Navbar />
</StyledPanel>
```

When it collapses, passing the 30%, it calls the function `onCollapse` and I can see `got here, collapsed and resize`, but it doesn't actually change the size of the panel to `10`.

**Comment by bvaughn** [comment]
> - Even tho we call resize and it's being set to `10`, the panel never changes size actually. You can see that when you go narrow than `SIDEBAR_MIN_SIZE`, we call `collapseSidePanel` which calls `sidebarPanel.resize(2);` and it seems to resize it, but because we are dragging it(i think), it never changes size.

I think this is the key here. There is no way to cancel the active drag, so the size you set is immediately overridden by the next move event.

I think this is something I could possibly probably add to the API. Let me think on it briefly.

**Comment by bvaughn** [comment]
Okay. I've given this a lot more thought and I think the cancel-drag API would be pretty awkward to work with. I think adding an optional `collapsedSize` prop (defaults to 0) would be the most reasonable approach.

**Comment by bvaughn** [comment]
Added in a87cdb6. Will release as 0.0.52

**Comment by bvaughn** [comment]
Published in 0.0.52!

Here's an updated sandbox for you:
https://codesandbox.io/s/bold-sky-7r4qk9?file=/src/App.js

**Comment by fidalgodev** [comment]
Awesome!! 🥳
